lib/btrfs_test.pm

```diff
@@ -114,6 +114,55 @@ sub cron_mock_lastrun {
     assert_script_run 'ls -al /var/spool/cron/lastrun/cron.{hourly,daily,weekly,monthly}';
 }
 
+# recent versions of snapper support json output
+# otherwise fallback to old implementation of parsing
+sub get_last_snap_number {
+    if (!defined($old_snapper)) {
+        $old_snapper = script_run('snapper --jsonout list');
+    }
+
+    if (!!$old_snapper) {
+        return _get_last_snap_number_old();
+    }
+
+    my $snaps = decode_json(script_output('snapper --jsonout list'));
```

Comment:
If you touch this anyway, you can use `list --disable-used-space`. That speeds it up significantly as it skips a qgroup scan.

Reply:
Updated

Reply:
Same here (and other places)

Reply:
oh yeah, added, thanks :)

Reply:
ok, I have to revert the `disable-used-space` option. It is not present in old snappers either
http://kepler.suse.cz/tests/23337#step/snapper_create/43